### DISCUSSION TITLE
Memory leak

### DISCUSSION THREAD

**Discussion by ivannkf** [vikejs/vike#653] [Discussion]
Hi!

I've been using vite-plugin-ssr for a while, and it's working quite well for us. The problem is that we have a memory leak in production and we can't find the root cause. The main retained objects are vue-router objects (we use it as our router currently) and also pageContext objects. Here are some screenshots of the heap allocation (there are many objects of each one of the screenshots):

![Screenshot 2023-02-15 at 16 05 23](https://user-images.githubusercontent.com/1236564/219066609-177ff0fb-55fd-4034-bef1-656102ef4b83.png)
![Screenshot 2023-02-15 at 16 05 31](https://user-images.githubusercontent.com/1236564/219066619-b82859d6-9993-423d-a44b-8b39027ac97a.png)
![Screenshot 2023-02-15 at 16 05 41](https://user-images.githubusercontent.com/1236564/219066622-d62e7fc8-da6f-411d-a8b6-f8fad7d0458a.png)

In the second one you can see that there is a retainer from     __VUE_DEVTOOLS_PLUGINS__, which as far as I know should not happen in prod. I double checked and the NODE_ENV is production, so not sure why it's there.

We use the code of the example (create a new router for every new app in every request). 

I've been trying to fix it for days now without any luck. Hopefully someone here can help out.

Thank you!

**Comment by brillout** [comment]
VPS's `pageContext` seems to have a low shallow size, which makes me think it's probably a user land problem. But let me know if you have evidence that it's related to VPS.

**Reply by ivannkf** [reply]
I created a repo to reproduce the the issue here: https://github.com/ivannkf/memory-leak-reproduction 

It doesn't happen without the i18n library, but I don't know if the issue is on their side, on VPS or if I'm actually doing something wrong.

**Reply by brillout** [reply]
> It doesn't happen without the i18n library

It's probably related to that library then. You can try to reproduce without using the library to be sure (in which case I'll dig into it).
